src/NHibernate/Util/EnumerableExtensions.cs

```diff
@@ -6,6 +6,16 @@ namespace NHibernate.Util
 {
 	public static class EnumerableExtensions
 	{
+		public static HashSet<T> ToHashSet<T>(this IEnumerable<T> collection)
```

Comment:
Indeed, its sole purpose is exploiting type inference for avoiding having to explicitly specify the element type by calling HashSet<> constructor.